lib/services/project-service.ts

```diff
@@ -68,25 +67,71 @@ export class ProjectService implements IProjectService {
 				this.$logger.trace("Copying custom app into %s", appDirectory);
 				appPath = customAppPath;
 			} else {
-				// No custom app - use nativescript hello world application
-				this.$logger.trace("Using NativeScript hello world application");
-				let defaultTemplatePath = this.$projectTemplatesService.defaultTemplatePath.wait();
-				this.$logger.trace("Copying NativeScript hello world application into %s", appDirectory);
+				let defaultTemplatePath = this.$projectTemplatesService.prepareTemplate(selectedTemplate).wait();
+				this.$logger.trace(`Copying application from '${defaultTemplatePath}' into '${appDirectory}'.`);
 				appPath = defaultTemplatePath;
 			}
 
 			try {
 				this.createProjectCore(projectDir, appPath, projectId).wait();
+				//update dependencies and devDependencies of newly created project with data from template
+				this.mergeProjectAndTemplateProperties(projectDir, appPath).wait();
+				this.updateAppResourcesDir(appDirectory).wait();
+				// Delete app/package.json file, its just causing confusion.
+				// Also its dependencies and devDependencies are already merged in project's package.json.
+				this.$fs.deleteFile(path.join(projectDir, constants.APP_FOLDER_NAME, constants.PACKAGE_JSON_FILE_NAME)).wait();
 			} catch (err) {
 				this.$fs.deleteDirectory(projectDir).wait();
 				throw err;
 			}
-
 			this.$logger.out("Project %s was successfully created", projectName);
 
 		}).future<void>()();
 	}
 
+	private mergeProjectAndTemplateProperties(projectDir: string, templatePath: string): IFuture<any> {
```

Comment:
You are not returning but the return type is Future\<any\>. Consider Future\<void\>